File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.cpp
#include <string>
#include <vector>
#include <queue>

using namespace std;



vector<int> solution(int k, vector<int> score) {
    priority_queue<int, vector<int>, greater<>> pq;
    vector<int> answer;
    for(const auto i : score){
        pq.push(i);
        if(pq.size() > k) pq.pop();
        answer.push_back(pq.top());
    }
    return answer;
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.js
function solution(k, score) {
    let answer = [];
    let hof = [];
    score.forEach(s => {
        hof.push(s);
        hof.sort((a,b) => b-a);
        if(hof.length > k) hof.pop();
        answer.push(hof.at(-1));
    })
    return answer;
}